### PR TITLE
Support OR for Rules in YAML

### PR DESCRIPTION
**Pull request by degiz**
**Proposed changes**:
- Support `or` key in YAML Rules

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by degiz** [comment]
@tmbo adding you in the review, but it's a tiny one 😄

File: data/test_yaml_stories/rules_without_stories.yml [hunk 1]
```diff
@@ -34,3 +34,12 @@ rules:
...
   - intent: explain
   - action: utter_explain_some_slot
+
+- rule: start optional preferences
+  steps:
+  - action: utter_request_optional
+  - or:
+    - intent: deny
+    - intent: inform
+  - action: form_plane_search_preference
```
Comment (Ghostvv) [inline data/test_yaml_stories/rules_without_stories.yml:44]: let's have some more common names in this test

Comment (degiz) [reply data/test_yaml_stories/rules_without_stories.yml:44]: what do you mean?

**Review by tmbo — APPROVED** [review]

File: data/test_yaml_stories/rules_without_stories.yml (hunk 1 above)
Comment (Ghostvv) [reply data/test_yaml_stories/rules_without_stories.yml:44]: it's weird to have these arbitrary concrete action names in a test

**Pull request merged**